nifi-nar-bundles/nifi-elasticsearch-5-bundle/nifi-elasticsearch-5-processors/src/main/java/org/apache/nifi/processors/elasticsearch/PutElasticsearch5.java

```diff
@@ -0,0 +1,266 @@
+        + "the index to insert into and the type of the document. If the cluster has been configured for authorization "
+        + "and/or secure transport (SSL/TLS), and the X-Pack plugin is available, secure connections can be made. This processor "
+        + "supports Elasticsearch 5.x clusters.")
+public class PutElasticsearch5 extends AbstractElasticsearch5TransportClientProcessor {
+
+    static final Relationship REL_SUCCESS = new Relationship.Builder().name("success")
+            .description("All FlowFiles that are written to Elasticsearch are routed to this relationship").build();
+
+    static final Relationship REL_FAILURE = new Relationship.Builder().name("failure")
+            .description("All FlowFiles that cannot be written to Elasticsearch are routed to this relationship").build();
+
+    static final Relationship REL_RETRY = new Relationship.Builder().name("retry")
+            .description("A FlowFile is routed to this relationship if the database cannot be updated but attempting the operation again may succeed")
+            .build();
+
+    public static final PropertyDescriptor ID_ATTRIBUTE = new PropertyDescriptor.Builder()
+            .name("el5-put-id-attribute")
+            .displayName("Identifier Attribute")
+            .description("The name of the attribute containing the identifier for each FlowFile")
+            .required(true)
+            .expressionLanguageSupported(false)
+            .addValidator(StandardValidators.ATTRIBUTE_KEY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor INDEX = new PropertyDescriptor.Builder()
+            .name("el5-put-index")
+            .displayName("Index")
+            .description("The name of the index to insert into")
+            .required(true)
+            .expressionLanguageSupported(true)
+            .addValidator(StandardValidators.createAttributeExpressionLanguageValidator(
+                    AttributeExpression.ResultType.STRING, true))
+            .build();
+
+    public static final PropertyDescriptor TYPE = new PropertyDescriptor.Builder()
+            .name("el5-put-type")
+            .displayName("Type")
+            .description("The type of this document (used by Elasticsearch for indexing and searching)")
+            .required(true)
+            .expressionLanguageSupported(true)
+            .addValidator(StandardValidators.createAttributeExpressionLanguageValidator(
+                    AttributeExpression.ResultType.STRING, true))
+            .build();
+
+    public static final PropertyDescriptor INDEX_OP = new PropertyDescriptor.Builder()
+            .name("el5-put-index-op")
+            .displayName("Index Operation")
+            .description("The type of the operation used to index (index, update, upsert)")
+            .required(true)
+            .expressionLanguageSupported(true)
+            .addValidator(StandardValidators.createAttributeExpressionLanguageValidator(
```

Comment:
This should do what "host" does for validation, if EL is present then mark valid else validate what is there (ie. check if it is index, update or upsert)